stdlib/public/SwiftShims/CMakeLists.txt

```diff
@@ -110,7 +110,8 @@ endif()
 # First extract the "version" used for Clang's resource directory.
 string(REGEX MATCH "[0-9]+\\.[0-9]+(\\.[0-9]+)?" CLANG_VERSION
   "${LLVM_PACKAGE_VERSION}")
-if(NOT SWIFT_INCLUDE_TOOLS AND SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER)
+if(NOT SWIFT_INCLUDE_TOOLS AND
+   (SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER OR SWIFT_PREBUILT_CLANG))
```

Comment:
@gottesmm and @shahmishal, should this be using the passed-in prebuilt compiler's resource directory even when `SWIFT_INCLUDE_TOOLS` is true, ie should this be `if(SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER OR SWIFT_PREBUILT_CLANG)` instead? This only applies when building the standalone stdlib now, but since you guys have started cross-compiling the full Mac toolchain with `SWIFT_NATIVE_CLANG_TOOLS_PATH` set (as do I when cross-compiling the Termux toolchain for Android), I'm not sure which resource directory should be used, the freshly built one that's now used or the one that comes with the prebuilt clang, or if it matters.

Reply:
This code path uses `CMAKE_C_COMPILER` which no longer is setup by build-script-impl now.  How is this intended to work?  It seems a bit confusing.

Reply:
It is still using it, it may help if you look at all three commits in this pull together.

I agree that all the `CMAKE_C_COMPILER` invocations are confusing, as [I explained in #33675 a couple weeks ago](https://github.com/apple/swift/pull/33675#issuecomment-716950099) (see that original comment for the links too):
```
The build-script requires a host_cc to be set,
which it looks up in the path or can be passed
in with --host-cc and then it sets CMAKE_{C,CXX}_COMPILER
for all CMake invocations with that host clang/clang++.
Then, several build products override that with
another CMAKE_{C,CXX}_COMPILER invocation,
either set to the freshly built Swift clang or native_clang_tools_path, #32922.

On top of this, you've now added a third CMAKE_{C,CXX}_COMPILER
invocation for all build-script-impl CMake invocations,
including when compiling CMark, LLVM, and Swift.
```
I'm only removing that third invocation he added in this pull, `CMAKE_C_COMPILER` is still set for the stdlib using `SWIFT_NATIVE_CLANG_TOOLS_PATH` in `stdlib/CMakeLists.txt` in this pull.